lib/puppet_pal.rb

```diff
@@ -190,8 +239,31 @@ def self.main(manifest = nil, facts = nil, &block)
 
         rescue => detail
           Puppet.log_exception(detail)
-          raise # exit(1)
+          raise
       end
     end
   end
+
+  def self.assert_non_empty_string(s, what, allow_nil=false)
+    unless s.is_a?(String) && s.length > 0 || (allow_nil && s.nil?)
```

Comment:
An one-line alternative with wording consistent with the rest of puppet could be to use type assertions, e.g.
```
TypeAsserter.assert_instance_of(nil, PStringType::NON_EMPTY, s, allow_nil) { _('Puppet Pal: %{what}') % {what: what} }
```

Reply:
I used this.